engine/src/main/java/org/terasology/engine/world/block/shapes/BlockMeshPart.java

```diff
@@ -94,8 +94,8 @@ public void appendTo(ChunkMesh chunk, ChunkView chunkView, int offsetX, int offs
             elements.color.put(colorOffset);
             elements.position.put(pos.set(vertices[vIdx]).add(offsetX, offsetY, offsetZ));
             elements.normals.put(normals[vIdx]);
-            elements.flags.put(flags.getValue());
-            elements.frames.put(texFrames - 1);
+            elements.flags.put((byte) (flags.getValue()));
+            elements.frames.put((byte) (texFrames - 1));
```

Comment:
you can verify this by starting CoreWorlds and giving a lava block and verifying the animations play correctly.